src/languages/typescript.ts

```diff
@@ -0,0 +1,63 @@
+import type { LanguagePattern } from "../types";
+
+
+export const Typescript: LanguagePattern[] = [{
+  pattern: /(Readonly<|ReadonlyArray<|Array<|Record<|Pick<|Omit<|Exclude<|Extract<)/,
```

Comment:
ideally we should look for patterns with `PascalCase<` instead of these limited identifiers, but ig that could potentially match some false positive cases

let's stick with this for now